src/network/networkd-address.c

```diff
@@ -20,6 +21,103 @@
 #define ADDRESSES_PER_LINK_MAX 2048U
 #define STATIC_ADDRESSES_PER_NETWORK_MAX 1024U
 
+int generate_ipv6_eui_64_address(Link *link, uint8_t *ret) {
+        uint8_t eui64[8];
+
+        assert(link);
+        assert(ret);
+
+        eui64[0]  = link->mac.ether_addr_octet[0];
```

Comment:
Why not just assign these values directly into ret?

I don't see why the eui64 array needs to exist.